app/boards/arm/dz60rgb/dz60rgb_rev1.yaml

```diff
@@ -1,5 +1,5 @@
-identifier: DZ60RGB_rev1
-name: DZ60RGBREV1
+identifier: dz60rgb_rev1
+name: DZ60RGB Rev1
 type: keyboard
```

Comment:
The `type` property here is defined here: https://docs.zephyrproject.org/latest/guides/test/twister.html#twister-script
`mcu` or `qemu` - for testing purposes.
I'd suggest we don't improvise on this field.

The directory structure indicates whether it's a board or shield.

Reply:
This type should be `mcu`, yes. My mistake from when I made this board long ago. I'm not so sure what the issue is with extending this to have `shield` on it. It makes this yaml much more consumable without needing path context.

Reply:
Technically we're breaking the inherited intention (schema) of that field by doing so.  A different, dedicated field which indicates whether it's a shield would be more appropriate.  None of the shields in Zephyr have such a field though - or the `type` field for that matter.

Reply:
I appreciate that none of the shields in Zephyr have a yaml file at all.

Reply:
The easy way to approach this is to have two globs.  One for boards, one for shields.

Reply:
We should consider whether every application that ever consumes these yaml files will have the full path context when working with them. I'm not sure we can guarantee that, so I think we should still have something indicating whether a piece of hardware is a shield.

We could remove the `type` entirely and check if it's undefined to tell if it's a shield, but I don't really get how that's breaking the twister script any less than using a type that doesn't exist for hardware that shouldn't even be run with twister.

Reply:
As discussed privately, a lot of the semantic overlap issues can be solved with our own `.zmk.yaml` files.  I'd suggest going in that directly with a bespoke (unwritten) data schema that meets our needs.